# Inferential Statistics

## One sample test

In one sample test, we compare the population parameter such as mean of a single sample of data collected from a single population.

### 2) t test

### Very rarely we know the variance of the population. 

A common strategy to assess hypothesis is to conduct a t test. A t test can tell whether two groups have the same mean. 
A t test can be estimated for:
* 1) One sample t test
* 2) Two sample t test (including paired t test)

We assume that the samples are randomly selected, independent and come from a normally distributed population with unknown but equal variances.

### One sample t test

In [4]:
from scipy.stats             import ttest_1samp,ttest_ind, wilcoxon
from statsmodels.stats.power import ttest_power
import matplotlib.pyplot     as     plt

### Example 2

Suppose that a doctor claims that 17 year olds have an average body temperature that is higher than the commonly accepted average human temperature of 98.6 degree F.

A simple random statistical sample of 25 people, each of age 17 is selected. 

| ID | Temperature |
| --- | ----- |
| 1 | 98.56 | 
| 2 | 98.66 |
| 3 | 97.54 |
| 4 | 98.71 |
| 5 | 99.22 |
| 6 | 99.49 |
| 7 | 98.14 |
| 8 | 98.84 |
| 9 | 99.28 |
| 10 | 98.48 |
| 11 | 98.88 |
| 12 | 97.29 |
| 13 | 98.88 |
| 14 | 99.07 |
| 15 | 98.81 |
| 16 | 99.49 |
| 17 | 98.57 |
| 18 | 97.98 |
| 19 | 97.75 |
| 20 | 97.69 |
| 21 | 99.28 |
| 22 | 98.52 |
| 23 | 98.82 |
| 24 | 98.81 |
| 25 | 98.22 |


In [5]:
temperature = np.array([98.56, 98.66, 97.54, 98.71, 99.22, 99.49, 98.14, 98.84,\
                         99.28, 98.48, 98.88, 97.29, 98.88, 99.07, 98.81, 99.49,\
                         98.57, 97.98, 97.75, 97.69, 99.28, 98.52, 98.82, 98.81, 98.22])            

In [6]:
print('Mean is %2.1f Sd is %2.1f' % (temperature.mean(),np.std(temperature,ddof = 1)))

Mean is 98.6 Sd is 0.6


### Step 1: Define null and alternative hypotheses

In testing whether 17 year olds have an average body temperature that is higher than 98.6 deg F,the null hypothesis states that mean bdy temperature, $\mu$ equals 98.6. The alternative hypthesis states that the mean body temprature, $\mu$ is greater  than 98.6.

* $H_0$: $\mu$ = 98.6
* $H_A$: $\mu$ > 98.6

### Step 2: Decide the significance level

Here we select $\alpha$ = 0.05 and it is given that n, sample size = 25.

### Step 3: Identify the test statistic

We do not know the population standard deviation and the sample is not a large sample, n < 30. So you use the t distribution and the $t_STAT$ test statistic.

### Step 4: Calculate the p - value and test statistic

**scipy.stats.ttest_1samp calculates the t test for the mean of one sample given the sample observations and  the expected value in the null hypothesis. This function returns t statistic and two-tailed p value.**

In [7]:
t_statistic, p_value = ttest_1samp(temperature, 98.9)

In [8]:
print(t_statistic, p_value)

-2.50739461327 0.0193331116224


### Step 5 Decide to reject or accept null hypothesis

In this example, p value is 0.0193 and it is less than 5% level of significance

So the statistical decision is to reject the null hypothesis at 5% level of significance.

##### So there is sufficient evidence  to prove that 17 year olds have an average body temperature that is higher than the commonly accepted average human temperature of 98.6 degree F.